pkgs/development/python-modules/wxPython/4.0.nix

```diff
@@ -1,35 +1,83 @@
 { lib
+, stdenv
+, openglSupport ? true
+, libX11
+, pyopengl
 , buildPythonPackage
 , fetchPypi
 , pkgconfig
 , gtk3
 , libjpeg
 , libtiff
+, wxmac
+, darwin
 , SDL
 , gst-plugins-base
 , libnotify
 , freeglut
 , xorg
 , which
+, cairo
+, pango
+, python3
+, doxygen
+, ncurses
+, libpng
+, gstreamer
+, wxGTK30
+, python3Packages
 }:
 
+let
+wxGTK = wxGTK30.override { withGtk2 = false; withWebKit = true; };
+in
 buildPythonPackage rec {
   pname = "wxPython";
   version = "4.0.4";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0d9ef4260cb2f3e23ed9dcf6baa905ba585ac7d631613cddc299c4c83463ae29";
+    sha256 = "0adfccscii4rqbfkqq9isv3mln5s0nlvmxnwv4zf5wxj1hkg97hd";
   };
 
-  nativeBuildInputs = [
-    pkgconfig
-  ];
+  doCheck = false;
 
-  buildInputs = [
-    gtk3 libjpeg libtiff SDL gst-plugins-base libnotify freeglut xorg.libSM
-    which
-  ];
+  nativeBuildInputs = [ pkgconfig which doxygen
+    ]
+    ++ (lib.optionals (!stdenv.isDarwin) [ wxGTK libX11 ])
+    ++ (lib.optionals stdenv.isDarwin [ wxmac darwin.apple_sdk.frameworks.Cocoa ]);
+
+  buildInputs = with python3Packages; [ gtk3 libjpeg libtiff SDL gst-plugins-base libnotify freeglut xorg.libSM ncurses
+      requests libpng gstreamer wxGTK
```

Comment:
This unconditionally includes `gtk3` and `wxGTK` on mac, but just above you were trying to use `wxmac`?

Reply:
good catch, I'll follow the pattern from above